### PR TITLE
[java] Stop parsing Java for CPD

### PR DESCRIPTION
**Pull request by oowekyala**
I just noticed, the CPD lexer now parsed java code. This is not necessary anymore since #4958. It is a big performance improvement.

## Related issues

- Fixes #

## Ready?

- [ ] Added unit tests for fixed bug/feature
- [ ] Passing all unit tests
- [ ] Complete build `./mvnw clean verify` passes (checked automatically by github actions)
- [ ] Added (in-code) documentation (if needed)

### CODE REVIEW

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/SyntacticJavaTokenizerFactory.java [hunk 1]
```diff
@@ -4,55 +4,24 @@
...
  */
+@Deprecated
 public final class SyntacticJavaTokenizerFactory {
     private SyntacticJavaTokenizerFactory() {
         // factory class
     }
 
+    @Deprecated
     public static TokenManager<JavaccToken> createTokenizer(CharStream cs) {
-        final List<JavaccToken> tokenList = new ArrayList<>();
```
Comment (adangel) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/SyntacticJavaTokenizerFactory.java:31]: We could leave this impl here, but just deprecate it and don't use it anymore.

**Review by adangel — COMMENTED** [review]
I'm moving this to 7.2.0 so that we don't need to rush the fix.

The unit test JavaCpdLexerTest is failing now. It seems that some tokens are missing now and other are not discarded anymore...

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to master:<br />
This changeset changes 0 violations,<br />
introduces 0 new violations, 0 new errors and 0 new configuration errors,<br />
removes 0 violations, 0 errors and 0 configuration errors.<br />
<a href="https://github.com/pmd/pmd/actions/runs/8869469543?pr=4973">Download full report as build artifact</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/SyntacticJavaTokenizerFactory.java (hunk 1 above)
Comment (oowekyala) [reply pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/SyntacticJavaTokenizerFactory.java:31]: Done

**Review by adangel — APPROVED** [review]

**Pull request merged**